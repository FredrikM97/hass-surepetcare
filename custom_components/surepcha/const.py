""" "Constants for the Sure Petcare integration."""

DOMAIN = "surepcha"

CONF_EMAIL = "email"
CONF_PASSWORD = "password"

TOKEN = "token"
CLIENT_DEVICE_ID = "client_device_id"
FACTORY = "surepetcare_factory"
TIMEOUT_API = 10
KEY_API = "surepetcare_api"
COORDINATOR_LIST = "coordinator_list"
COORDINATOR_DICT = "coordinator_dict"
COORDINATOR = "coordinator"
ENTRY_ID = "entry_id"
SCAN_INTERVAL = 300
POLLING_SPEED = "polling_speed"
LOCATION_INSIDE = "location_inside"
LOCATION_OUTSIDE = "location_outside"
OPTION_DEVICES = "devices"
DEVICE_OPTION = "device_option"
OPTIONS_FINISHED = "finished"
PRODUCT_ID = "product_id"
DEVICES = "devices"
NAME = "name"
