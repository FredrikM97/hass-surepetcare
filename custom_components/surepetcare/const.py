"""TODO."""

DOMAIN = "surepetcare"

CONF_EMAIL = "email"
CONF_PASSWORD = "password"

TOKEN = "token"
CLIENT_DEVICE_ID = "client_device_id"
FACTORY = "surepetcare_factory"
TIMEOUT_API = 10
KEY_API = "surepetcare_api"
COORDINATOR_LIST = "coordinator_list"
COORDINATOR_DICT = "coordinator_dict"
COORDINATOR = "coordinator"
