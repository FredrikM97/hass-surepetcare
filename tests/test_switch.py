import pytest
from unittest.mock import patch
from syrupy.assertion import SnapshotAssertion

from homeassistant.const import Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from . import initialize_entry

from pytest_homeassistant_custom_component.common import (
    MockConfigEntry,
    snapshot_platform,
)


@patch("custom_components.surepetcare.PLATFORMS", [Platform.SWITCH])
@pytest.mark.usefixtures("enable_custom_integrations")
@pytest.mark.usefixtures("entity_registry_enabled_default")
@pytest.mark.asyncio
async def test_platform_setup_and_discovery(
    hass: HomeAssistant,
    mock_client,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
    mock_devices,
    mock_pets,
) -> None:
    await initialize_entry(
        hass, mock_client, mock_config_entry, mock_devices, mock_pets
    )
    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)


@patch("custom_components.surepetcare.PLATFORMS", [Platform.SWITCH])
@pytest.mark.usefixtures("enable_custom_integrations")
@pytest.mark.usefixtures("entity_registry_enabled_default")
@pytest.mark.asyncio
async def test_platform_setup_and_discovery_missing_entities(
    hass: HomeAssistant,
    mock_client,
    mock_config_entry_missing_entities: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
    mock_devices,
    mock_pets,
) -> None:
    await initialize_entry(
        hass, mock_client, mock_config_entry_missing_entities, mock_devices, mock_pets
    )
    await snapshot_platform(
        hass, entity_registry, snapshot, mock_config_entry_missing_entities.entry_id
    )


@patch("custom_components.surepetcare.PLATFORMS", [Platform.SWITCH])
@pytest.mark.usefixtures("enable_custom_integrations")
@pytest.mark.usefixtures("entity_registry_enabled_default")
@pytest.mark.asyncio
async def test_switch_toggle_and_snapshot(
    hass: HomeAssistant,
    mock_client,
    mock_config_entry: MockConfigEntry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
    mock_devices,
    mock_pets,
) -> None:
    await initialize_entry(
        hass, mock_client, mock_config_entry, mock_devices, mock_pets
    )
    await hass.async_block_till_done()

    # Iterate over all switch entities
    for entity_id in hass.states.async_entity_ids("switch"):
        # Turn on the switch
        await hass.services.async_call(
            "switch",
            "turn_on",
            {"entity_id": entity_id},
            blocking=True,
        )
        await hass.async_block_till_done()
        state_on = hass.states.get(entity_id)
        assert state_on is not None
        snapshot(name=f"{entity_id}-on").assert_match(state_on)

        # Turn off the switch
        await hass.services.async_call(
            "switch",
            "turn_off",
            {"entity_id": entity_id},
            blocking=True,
        )
        await hass.async_block_till_done()
        state_off = hass.states.get(entity_id)
        assert state_off is not None
        snapshot(name=f"{entity_id}-off").assert_match(state_off)