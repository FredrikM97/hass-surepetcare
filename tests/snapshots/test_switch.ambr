# serializer version: 1
# name: test_platform_setup_and_discovery[switch.laundry_door_curfew_active-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.laundry_door_curfew_active',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Curfew active',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'curfew_enabled',
    'unique_id': '727608-curfew_enabled',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery[switch.laundry_door_curfew_active-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Laundry Door Curfew active',
    }),
    'context': <ANY>,
    'entity_id': 'switch.laundry_door_curfew_active',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_platform_setup_and_discovery[switch.maui_indoor_only-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.maui_indoor_only',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:door',
    'original_name': 'Indoor Only',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'indoor_only',
    'unique_id': '472721-indoor_only',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery[switch.maui_indoor_only-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui Indoor Only',
      'icon': 'mdi:door',
    }),
    'context': <ANY>,
    'entity_id': 'switch.maui_indoor_only',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
# name: test_platform_setup_and_discovery[switch.pet_with_empty_status_indoor_only-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.pet_with_empty_status_indoor_only',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:door',
    'original_name': 'Indoor Only',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'indoor_only',
    'unique_id': '222222-indoor_only',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery[switch.pet_with_empty_status_indoor_only-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Pet with empty status Indoor Only',
      'icon': 'mdi:door',
    }),
    'context': <ANY>,
    'entity_id': 'switch.pet_with_empty_status_indoor_only',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[switch.laundry_door_curfew_active-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.laundry_door_curfew_active',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': 'Curfew active',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'curfew_enabled',
    'unique_id': '727608-curfew_enabled',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[switch.laundry_door_curfew_active-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Laundry Door Curfew active',
    }),
    'context': <ANY>,
    'entity_id': 'switch.laundry_door_curfew_active',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[switch.pet_with_empty_status_indoor_only-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.pet_with_empty_status_indoor_only',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:door',
    'original_name': 'Indoor Only',
    'platform': 'surepetcare',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': 0,
    'translation_key': 'indoor_only',
    'unique_id': '222222-indoor_only',
    'unit_of_measurement': None,
  })
# ---
# name: test_platform_setup_and_discovery_missing_entities[switch.pet_with_empty_status_indoor_only-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Pet with empty status Indoor Only',
      'icon': 'mdi:door',
    }),
    'context': <ANY>,
    'entity_id': 'switch.pet_with_empty_status_indoor_only',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
# name: test_switch_toggle_and_snapshot[switch.laundry_door_curfew_active-off]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Laundry Door Curfew active',
    }),
    'context': <ANY>,
    'entity_id': 'switch.laundry_door_curfew_active',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_switch_toggle_and_snapshot[switch.laundry_door_curfew_active-on]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Laundry Door Curfew active',
    }),
    'context': <ANY>,
    'entity_id': 'switch.laundry_door_curfew_active',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_switch_toggle_and_snapshot[switch.maui_indoor_only-off]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui Indoor Only',
      'icon': 'mdi:door',
    }),
    'context': <ANY>,
    'entity_id': 'switch.maui_indoor_only',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
# name: test_switch_toggle_and_snapshot[switch.maui_indoor_only-on]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Maui Indoor Only',
      'icon': 'mdi:door',
    }),
    'context': <ANY>,
    'entity_id': 'switch.maui_indoor_only',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
# name: test_switch_toggle_and_snapshot[switch.pet_with_empty_status_indoor_only-off]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Pet with empty status Indoor Only',
      'icon': 'mdi:door',
    }),
    'context': <ANY>,
    'entity_id': 'switch.pet_with_empty_status_indoor_only',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
# name: test_switch_toggle_and_snapshot[switch.pet_with_empty_status_indoor_only-on]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'Pet with empty status Indoor Only',
      'icon': 'mdi:door',
    }),
    'context': <ANY>,
    'entity_id': 'switch.pet_with_empty_status_indoor_only',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'unavailable',
  })
# ---
